### PR TITLE
Re-enable Tom Hanks page, allow new pages to be accessed directly

### PR DESCRIPTION
**Pull request by forabi**
* [x] Allow new pages to be accessed directly using new slugs
    * [x] Move `redirectionMap.ts`, `isNewSlug.ts`, redirection logic to the new app directory (`src/app`) so that we can use the redirection map to set the canonical URL and use the generated API types for `isNewSlug.ts`.
    * [x] Make sure the canonical URLs of pages that are not explicitly redirected is set to the old website's page URL
    * [x] Make sure the canonical URLs of pages that are explicitly redirected is set the new website's page URL
  * [x] Re-enable redirection for Tom Hanks's page
  * [x] Test that the redirection works
  * [x] Test that calling `next()` in server middleware falls back to the old website
  * [x] Allow whitelisted pages to be indexed
  * [x] Fix duplicated `<head>` tags

### CODE REVIEW

**Comment by forabi** [comment]
@wholesomedev Want to take a look? With #271 merged, I don't think there remains any issues blocking launch. This PR:

* Redirects Tom Hanks's page to the new app
* Sets the canonical URL of redirected pages to the new app URLs
* Allows new pages to be indexed _only if they are whitelisted for redirection_
* Allows accessing of new app pages directly via URL
* Prevents indexing of new app pages that are not explicitly whitelisted
* Uses the old website page URL as the canonical URLs for new pages that are not explicitly whitelisted

File: src/app/redirectionMap.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ */
+export const redirectionMap = new Map<string, string>([
+  ['/tom-hanks', '/Tom_Hanks'],
+]);
+
+export const whitelistedNewPaths = [
+  ...Array.from(redirectionMap.values()),
+  '/search',
+  // tslint:disable-next-line:no-unnecessary-callback-wrapper
+].map(path => pathToRegExp(path));
```
Comment (wholesomedev) [inline src/app/redirectionMap.ts:15]: Why can't you just do `map(pathToRegExp)`?

**Review by wholesomedev — APPROVED** [review]
LGTM! 👍

File: src/app/redirectionMap.ts (hunk 1 above)
Comment (forabi) [reply src/app/redirectionMap.ts:15]: TypeScript says the signature/number of arguments does not match

**Pull request merged**